src/client.ts

```diff
@@ -231,18 +234,30 @@ class OptimizelyReactSDKClient implements ReactSDKClient {
       return { success: true };
     });
 
-    this._client.onReady().then(() => {
-      this.isClientReady = true;
-    });
-
-    this.dataReadyPromise = Promise.all([this.userPromise, this._client.onReady()]).then(() => {
-      // Client and user can become ready synchronously and/or asynchronously. This flag specifically indicates that they became ready asynchronously.
-      this.isReadyPromiseFulfilled = true;
-      return {
-        success: true,
-        reason: 'datafile and user resolved',
-      };
-    });
+    if (this._client) {
+      this._client.onReady().then(() => {
+        this.isClientReady = true;
+      });
+
+      this.dataReadyPromise = Promise.all([this.userPromise, this._client!.onReady()]).then(() => {
+        // Client and user can become ready synchronously and/or asynchronously. This flag specifically indicates that they became ready asynchronously.
+        this.isReadyPromiseFulfilled = true;
+        return {
+          success: true,
+          message: 'Successfully resolved datafile and user information.',
+        };
+      });
+    } else {
+      logger.warn('Unable to resolve datafile and user information because Optimizely client failed to initialize.');
+
+      this.dataReadyPromise = new Promise((resolve, reject) => {
```

Comment:
You could use `Promise.resolve(value)`
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/resolve

Reply:
Thanks @fabb - we'll implement that change in a future push!